README.md

```diff
@@ -1 +1,130 @@
-# earthstar-fetch
\ No newline at end of file
+# earthstar-fetch
+
+A `fetch()` interface for talking to EarthStar
+
+## Goals
+
+- Stick to existing conventions in HTTP-land (body, methods, prefer standard headers)
+- GET data from earthstar workspaces
+- GET an author address for a given name
+- PUT data into a workspace
+- Auto-generate identity keys from a 4 character name
+- Search via querystring params and path prefixes
+
+## Dream API
+
+Basic URL format:
+
+```
+earthstar://+gardening.xxxxxxxxx/path/to/key.json
+```
+
+GET document
+
+```JavaScript
+const res = await fetch('earthstar://+gardening.xxxxx/example')
+
+// Get the timestamp for when it was last updated
+res.headers.get('mtime') === "1231231231231"
+// Get the author for this document
+res.headers.get('earthstar-author') === '@bbbb.xxxxxxxx'
+// Get the signature for this document, in case you want it 🤷
+res.headers.get('earthstar-signature') === 'xxxxxxx'
+
+// Get the content of the document back out.
+// If it's JSON, you should use `.json()`
+const content = await res.text()
+```
+
+GET an identity 'name' - Generate an identity
+
+```JavaScript
+// Specify `@XXXX` where `X` is a lowercase ascii character
+// This will generate and store a keypair for this identity
+const res = fetch('earthstar://@xmpl')
+
+// Get the generated address for your identity
+const {address} = await res.json()
+
+address === '@xmpl.xxxxxxx'
+```
+
+POST some data, sign with identity
+
+```JavaScript
+// It's good practice to use a file extension so that the proper mime type gets set
+const res = await fetch('earthstar://+gardening.xxxxx/example.json', {
+  method: 'post',
+  body: JSON.stringify({
+    hello: 'world'
+  }),
+  headers: {
+    // Use the identity address you got before to sign your update
+    "earthstar-author": "@xmpl.xxxxxx"
+  }
+})
+
+// Get the timestamp for when it was last updated
+res.headers.get('mtime') === "1231231231231"
+// Your author address
+res.headers.get('earthstar-author') === '@xmpl.xxxxxxxx'
+// Get the signature for this document, in case you want it 🤷
+res.headers.get('earthstar-signature') === 'xxxxxxx'
+```
+
+POST some data scoped to your identity (others cant update)
+
+```JavaScript
+// Note the `/~@xml,
+const res = await fetch('earthstar://+gardening.xxxxx/example/~@xmpl.xxxxx/profile.json', {
+  method: 'post',
+  body: JSON.stringify({
+    name: 'Example'
+  }),
+  headers: {
+    // Use the identity address you got before to sign your update
+    "earthstar-author": "@xmpl.xxxxxx"
+  }
+})
+```
+
+Search - A get request on a prefix with querystring parameters
+
+Search options can be placed in the queryString section of the URL, more info can be found in the [QueryOpts interface](https://github.com/earthstar-project/earthstar/blob/master/src/util/types.ts#L194).
```

Comment:
FYI, The query options have changed a bit in the beta version of Earthstar

https://github.com/earthstar-project/earthstar/blob/beta/src/storage/query.ts#L28-L97

Reply:
...and changed one more time to get rid of snake_case, I think they will be stable now.

https://github.com/earthstar-project/earthstar/releases/tag/v6.1.0